### PR TITLE
Mininal version of datagrid operator with Infinispan cluster and cache

### PR DESCRIPTION
**Pull request by mvmaestri**
Minimal version of datagrid operator for OpenShift

RUNNING DATA GRID ON OPENSHIFT
https://access.redhat.com/documentation/en-us/red_hat_data_grid/8.1/html/running_data_grid_on_openshift/rhdg

### CODE REVIEW

**Comment by ckavili** [comment]
I was wondering if we can pull our [operatorhub](https://github.com/redhat-cop/helm-charts/tree/master/charts/operatorhub) chart as a dependency in `Chart.yaml` and overwrite the values, and has other objects like cache in the template only..what do you think? 🤔

cc: @springdo

File: charts/datagrid-operator/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+description: A Helm chart for customising the deployment of the datagrid operator ⚓️
+name: datagrid-operator
+type: application
+version: 0.0.1
+appVersion: 0.0.2
+home: https://github.com/redhat-cop/helm-charts
+icon: https://www.redhat.com/cms/managed-files/data-grid-slice2-500x415.png
+maintainers:
+- name: mvmaestri
+- name: fwalker@redhat.com
```
Comment (ckavili) [inline charts/datagrid-operator/Chart.yaml:11]: linting expects a new line at the end of the file 🤷

**Comment by eformat** [comment]
@mvmaestri , @ckavili, @springdo  - i'm sort of in two minds about this chart

ideally we could add in the hooks so that on an empty cluster we dont have the race condition between the operator subscription and the CR's when installing the chart.

also, most infra charts would wrap this as a dependency as a composite chart.. i.e. they would likely only want the subscription and operatorgroup .. as creating the cache/other objects is highly contextual based on the the app being written. i get its a minimal example .. but i wonder if there is a better way ? 

thoughts ?

File: charts/datagrid-operator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# operator manages upgrades etc
+operator:
+  name: datagrid
+  version: datagrid-operator.v8.1.2
+  channel: 8.1.x
+  operatorgroup: true
+
+
+# role bindings, enable this to restrict to listed namespaces only. cluster role by default (enabled: false)
+namespaceRoleBinding:
```
Comment (garethahealy) [inline charts/datagrid-operator/values.yaml:22]: is this used?

File: charts/datagrid-operator/values.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# control used by Labs Ubiquitous Journey
+enabled: true
+name: labs-datagrid
+
+ci_cd_namespace: &ci_cd "labs-ci-cd"
+dev_namespace: &dev "labs-dev"
+test_namespace: &test "labs-test"
+
+namespace: *ci_cd
+instancelabel: rht-labs.com/ubiquitous-journey
```
Comment (garethahealy) [inline charts/datagrid-operator/values.yaml:11]: is this used?

File: charts/datagrid-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+# control used by Labs Ubiquitous Journey
+enabled: true
+name: labs-datagrid
```
Comment (garethahealy) [inline charts/datagrid-operator/values.yaml:4]: is this used?

File: charts/datagrid-operator/values.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+ci_cd_namespace: &ci_cd "labs-ci-cd"
+dev_namespace: &dev "labs-dev"
+test_namespace: &test "labs-test"
```
Comment (garethahealy) [inline charts/datagrid-operator/values.yaml:8]: are theses used?

**Comment by garethahealy** [comment]
I agree with @eformat in that what the chart currently does, requires the user to change the templates to their needs for this to be usable. 

but at the same time, it provides a "quick helloworld" cluster but should that live here? or be doc'ed somewhere else? for example, there's a lot of stuff in container-quickstarts/pipelines repos which I don't think anyone knows what they are used for or by whom, don't think we want this repo heading in that direction.

**Comment by ckavili** [comment]
This was created for a possible use in one of our residencies..yeah, I see your concerns..maybe we can create a repo under rht-labs org like chart-we-like type thing and store them there. For example, we also created some charts for generic apps or Grafana/Prometheus configs which we may reuse. We can store them there 🤷

**Comment by eformat** [comment]
@ckavili @mvmaestri - i'm going to close this for now. A good home for it may be somewhere like here:

https://github.com/rht-labs/refactored-adventure

**Pull request closed without merging**